### PR TITLE
DBOS Class Docs

### PR DESCRIPTION
**Pull request by kraftp**
- Document main `dbos.DBOS` class
- Update code examples
- "AI-Powered Slackbot"

### CODE REVIEW

File: docs/python/reference/dbos-class.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+sidebar_position: 1
+title: DBOS Class
+---
+
+The DBOS class is a singleton&mdash;you may only instantiate it once in a program's lifetime.
```
Comment (qianl15) [inline docs/python/reference/dbos-class.md:6]: "initiate" is a bit vague. We can say "you may only call its constructor exactly once in a program's lifetyle"

Comment (qianl15) [reply docs/python/reference/dbos-class.md:6]: If I understand it correctly, you always have to call `DBOS()` even if you're explicitly using it?

File: docs/python/reference/dbos-class.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+```python
+DBOS(
+    *,
+    fastapi: Optional[FastAPI] = None,
+    config: Optional[ConfigFile] = None,
+)
+```
+
+**Parameters:**
+- `fastapi`: If your application is using FastAPI, the `FastAPI` object. If this is passed in, DBOS automatically calls `launch` when FastAPI is fully initialized. DBOS also adds to all routes a middleware that enables [tracing](../tutorials/logging-and-tracing.md#tracing) through FastAPI HTTP endpoints.
```
Comment (qianl15) [inline docs/python/reference/dbos-class.md:21]: Provide a pointer to "launch". It's unclear which launch it calls, FastAPI or itself.

File: docs/python/reference/dbos-class.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- `config`: A configuration object, which if passed in is used in place of `dbos-config.yaml`. We recommend using this for testing only.
```
Comment (qianl15) [inline docs/python/reference/dbos-class.md:22]: Explicitly say: we use dbos-config.yaml file by default.

File: docs/python/reference/dbos-class.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+# Define a scheduled workflow
+@DBOS.scheduled("* * * * *")
+@DBOS.workflow()
+def run_every_minute(scheduled_time: datetime, actual_time: datetime):
+    DBOS.logger.info("This is a scheduled workflow!")
+
+# In your main function, launch DBOS
+if __name__ == "__main__":
+    dbos.launch()
```
Comment (qianl15) [inline docs/python/reference/dbos-class.md:50]: It does look weird that we use `dbos.launch()` but `DBOS.destroy()`. This small dbos vs big DBOS still look weird. Why can't we just use `DBOS.launch()`?

File: docs/python/reference/dbos-class.md (hunk 1 above)
Comment (kraftp) [reply docs/python/reference/dbos-class.md:6]: Fixed

File: docs/python/reference/dbos-class.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+Launch DBOS, initializing database connections and starting scheduled workflows.
+Should be called in your main function.
+If a FastAPI app is passed into the `DBOS` constructor, `launch` is called automatically during FastAPI setup.
+
+**Example:**
+```python
+from dbos import DBOS
+
+# Initialize the DBOS object
+dbos = DBOS()
```
Comment (qianl15) [inline docs/python/reference/dbos-class.md]: No small dbos

**Review by qianl15 — APPROVED** [review]

**Pull request merged**